src/ol_infrastructure/infrastructure/mongodb/__main__.py

```diff
@@ -0,0 +1,209 @@
+salt_environment = Config("saltstack").get("environment_name") or environment_name
+instance_nums = range(mongodb_config.get_int("instance_count") or 3)
+
+for instance_num, subnet in zip(instance_nums, subnets):
+    instance_name = f"mongodb-{environment_name}-{instance_num}"
+    salt_minion = OLSaltStackMinion(
+        f"saltstack-minion-{instance_name}",
+        OLSaltStackMinionInputs(minion_id=instance_name),
+    )
+
+    cloud_init_userdata = build_userdata(
+        instance_name=instance_name,
+        minion_keys=salt_minion,
+        minion_roles=["mongodb"],
+        minion_environment=salt_environment,
+        salt_host=f"salt-{stack_info.env_suffix}.private.odl.mit.edu",
+        additional_cloud_config={
+            "fs_setup": [
+                {
+                    "device": "/dev/nvme0n1",
+                    "filesystem": "ext4",
+                    "label": "var_lib_mongodb",
+                    "partition": "None",
+                }
+            ],
+            "mounts": [
+                [
+                    "LABEL=var_lib_mongodb",
+                    "/var/lib/mongodb",
+                    "ext4",
+                    "defaults",
+                    "0",
+                    "0",
+                ]
+            ],
+        },
+    )
+
+    instance_tags = aws_config.merged_tags(
+        {
+            "Name": instance_name,
+            "mongodb_env": environment_name,
+            "escluster": salt_environment,
+        }
+    )
+    mongodb_security_groups = [
+        destination_vpc["security_groups"]["salt_minion"],
+        consul_stack.require_output("security_groups")["consul_agent"],
+        mongodb_security_group.id,
+    ]
+    mongodb_instance = ec2.Instance(
+        f"mongodb-instance-{environment_name}-{instance_num}",
+        ami=mongodb_config.get("ami_id") or debian_10_ami.id,
+        user_data=cloud_init_userdata,
+        instance_type=instance_type,
+        iam_instance_profile=mongodb_instance_profile.id,
+        tags=instance_tags,
+        volume_tags=instance_tags,
+        subnet_id=subnet,
+        key_name=salt_config.require("key_name"),
+        root_block_device=ec2.InstanceRootBlockDeviceArgs(
+            volume_type="gp2",
+            volume_size=20,  # noqa: WPS432
+            encrypted=True,
+        ),
+        ebs_block_devices=[
+            ec2.InstanceEbsBlockDeviceArgs(
+                device_name="/dev/sdb",
+                volume_type="gp2",
+                volume_size=mongodb_config.get_int("disk_size") or 100,
+                encrypted=True,
+            )
+        ],
+        vpc_security_group_ids=mongodb_security_groups,
+        opts=ResourceOptions(depends_on=[salt_minion]),
+    )
+    mongodb_instances.append(mongodb_instance)
+
+    export_data[instance_name] = {
+        "public_ip": mongodb_instance.public_ip,
+        "private_ip": mongodb_instance.private_ip,
+        "ipv6_address": mongodb_instance.ipv6_addresses,
+    }
```

Comment:
This is missing a call to `pulumi.export` at the end of the file to actually export the `export_data` contents as part of the stack output